lib/active_merchant/billing/gateways/rapyd.rb

```diff
@@ -0,0 +1,243 @@
+require 'pry'
+require 'pry-byebug'
+
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class RapydGateway < Gateway
+      self.test_url = 'https://sandboxapi.rapyd.net/v1/'
+      self.live_url = 'https://api.rapyd.net/v1/'
+
+      self.supported_countries = %w(US BR CA CL CO DO SV MX PE PT VI AU HK IN ID JP MY NZ PH SG KR TW TH VN AD AT BE BA BG HR CY CZ DK EE FI FR GE DE GI GR GL HU IS IE IL IT LV LI LT LU MK MT MD MC ME NL GB NO PL RO RU SM SK SI ZA ES SE CH TR VA)
+      self.default_currency = 'USD'
+      self.supported_cardtypes = %i[visa master american_express discover]
+
+      self.homepage_url = 'https://www.rapyd.net/'
+      self.display_name = 'Rapyd Gateway'
+
+      STANDARD_ERROR_CODE_MAPPING = {}
+
+      def initialize(options = {})
+        requires!(options, :secret_key, :access_key)
+        super
+      end
+
+      def purchase(money, payment, options = {})
+        post = {}
+        add_invoice(post, money, options)
+        add_payment(post, payment, options)
+        add_address(post, payment, options)
+        post[:capture] = true if payment_is_card?(options)
+
+        if payment_is_ach?(options)
+          MultiResponse.run do |r|
+            r.process { commit(:post, 'payments', post) }
+            post = {}
+            post[:token] = r.authorization
+            post[:param2] = r.params.dig('data', 'original_amount').to_s
+            r.process { commit(:post, 'payments/completePayment', post) }
+          end
+        else
+          commit(:post, 'payments', post)
+        end
+      end
+
+      def authorize(money, payment, options = {})
+        post = {}
+        add_invoice(post, money, options)
+        add_payment(post, payment, options)
+        add_address(post, payment, options)
+        post[:capture] = false
+        commit(:post, 'payments', post)
+      end
+
+      def capture(money, authorization, options = {})
+        post = {}
+        commit(:post, "payments/#{authorization}/capture", post)
+      end
+
+      def refund(money, authorization, options = {})
+        post = {}
+        post[:payment] = authorization
+        add_invoice(post, money, options)
+        commit(:post, 'refunds', post)
+      end
+
+      def void(authorization, options = {})
+        post = {}
+        commit(:delete, "payments/#{authorization}", post)
+      end
+
+      # Gateway returns an error if trying to run a $0 auth as invalid payment amount
+      # Gateway does not support void on a card transaction and refunds can only be done on completed transactions (such as a purchase). Authorize transactions are considered 'active' and not 'complete'  until they are captured.
```

Comment:
Thank you for the nice comment here. The flow is atypical and I think the explanation is warranted. Your logic makes sense to me. 👍 

Do you know if the merchant(s) requesting integration with this gateway are aware of this quirk of our implementation? Namely, that a `verify` call will in fact charge a card a set amount and then void that charge? It might be worth passing this info along to them because I think running a `purchase` here is not typical and not something a merchant would expect from `verify`.